### PR TITLE
Support Sparse values in index API

### PR DESCRIPTION
**Pull request by acatav**
Add support for sparse values in index API and grpc index API.
We are going to add tests in the next iteration

### CODE REVIEW

File: pinecone/core/client/model/index_meta_database.py [hunk 1]
```diff
@@ -101,6 +101,7 @@ def openapi_types():
             'shards': (int,),  # noqa: E501
             'pod_type': (str,),  # noqa: E501
             'index_config': (dict,),  # noqa: E501
+            'metadata_config': ({str: (bool, date, datetime, dict, float, int, list, str, none_type)},),  # noqa: E501
```
Comment (yarden-slon) [inline pinecone/core/client/model/index_meta_database.py:104]: why is it here?

File: pinecone/core/client/model/query_request.py [hunk 2]
```diff
@@ -179,6 +183,7 @@ def _from_openapi_data(cls, top_k, *args, **kwargs):  # noqa: E501
             include_metadata (bool): Indicates whether metadata is included in the response as well as the ids.. [optional] if omitted the server will use the default value of False  # noqa: E501
             queries ([QueryVector]): DEPRECATED. The query vectors. Each `query()` request can contain only one of the parameters `queries`, `vector`, or  `id`.. [optional]  # noqa: E501
             vector ([float]): The query vector. This should be the same length as the dimension of the index being queried. Each `query()` request can contain only one of the parameters `id` or `vector`.. [optional]  # noqa: E501
+            sparse_vector (SparseValues): [optional]  # noqa: E501
```
Comment (yarden-slon) [inline pinecone/core/client/model/query_request.py:186]: Can there be a description like there is for any other field?

File: pinecone/core/grpc/index_grpc.py [hunk 3]
```diff
@@ -487,14 +501,16 @@ def _query_transform(item):
         if filter is not None:
             filter = dict_to_proto_struct(filter)
 
+        sparse_vector = self._parse_sparse_values_arg(sparse_vector)
```
Comment (yarden-slon) [inline pinecone/core/grpc/index_grpc.py:504]: Why about _query_transform? should you handle sparse_values there too?

File: pinecone/core/grpc/index_grpc.py [hunk 4]
```diff
@@ -266,9 +267,13 @@ def upsert(self,
         If a new value is upserted for an existing vector id, it will overwrite the previous value.
 
         Examples:
-            >>> index.upsert([('id1', [1.0, 2.0, 3.0], {'key': 'value'}), ('id2', [1.0, 2.0, 3.0])], namespace='ns1')
+            >>> index.upsert([('id1', [1.0, 2.0, 3.0], {'key': 'value'}), ('id2', [1.0, 2.0, 3.0])],
```
Comment (yarden-slon) [inline pinecone/core/grpc/index_grpc.py:270]: Where are the sparse values in upsert?

File: pinecone/index.py [hunk 5]
```diff
@@ -77,22 +78,35 @@ def upsert(self,
 
         Examples:
             >>> index.upsert([('id1', [1.0, 2.0, 3.0], {'key': 'value'}), ('id2', [1.0, 2.0, 3.0])])
-            >>> index.upsert([Vector(id='id1', values=[1.0, 2.0, 3.0], metadata={'key': 'value'}),
-            >>>              Vector(id='id2', values=[1.0, 2.0, 3.0])])
+            >>> index.upsert([Vector(id='id1',
```
Comment (yarden-slon) [inline pinecone/index.py:81]: please add sparse values to upsert

File: pinecone/index.py [hunk 6]
```diff
@@ -250,6 +264,7 @@ def query(self,
               filter: Optional[Dict[str, Union[str, float, int, bool, List, dict]]] = None,
               include_values: Optional[bool] = None,
               include_metadata: Optional[bool] = None,
+              sparse_vector: Optional[Dict[str, Union[List[float], List[int]]]] = None,
```
Comment (yarden-slon) [inline pinecone/index.py:267]: I think it should be an option to get a SparseValues object and not just a dict

File: pinecone/core/client/model/index_meta_database.py (hunk 1 above)
Comment (acatav) [reply pinecone/core/client/model/index_meta_database.py:104]: we checked and it should

File: pinecone/index.py [hunk 7]
```diff
@@ -305,25 +326,17 @@ def _query_transform(item):
...
         args_dict = self._parse_non_empty_args([('vector', vector),
                                                 ('id', id),
                                                 ('queries', queries),
                                                 ('top_k', top_k),
                                                 ('namespace', namespace),
                                                 ('filter', filter),
                                                 ('include_values', include_values),
-                                                ('include_metadata', include_metadata)])
-
-        def _query_transform(item):
```
Comment (yarden-slon) [inline pinecone/index.py:317]: Why did you delete this? It's still in use (line 307)
